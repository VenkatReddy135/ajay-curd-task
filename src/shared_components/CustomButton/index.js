/* eslint-disable react/destructuring-assignment */
import React from "react";
import PropTypes from "prop-types";

/**
 *
 * @param {object} props all props
 *@function CustomButton
 * @returns {React.createElement} CustomButton
 */
export default function CustomButton(props) {
  return (
    <button onClick={props.handleClick} className={props.buttonClass}>
      {props.buttonValue}
    </button>
  );
}

CustomButton.propTypes = {
  buttonClass: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
  buttonValue: PropTypes.string,
};

CustomButton.defaultProps = {
  buttonValue: "",
  handleClick: () => {},
};
