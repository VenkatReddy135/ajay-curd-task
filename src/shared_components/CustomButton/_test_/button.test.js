import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import CustomButton from "../index";
import { cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-17-updated";
Enzyme.configure({ adapter: new Adapter() });

let wrapper;

beforeEach(() => {
  wrapper = shallow(<CustomButton />);
});
const props = {
  buttonClass: "",
  handleClick: () => {},
  buttonValue: "",
};
describe("Button Component Testing", () => {
  it("renders component", () => {
    const tree = renderer.create(<CustomButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CustomButton {...props} />, div);
  });
  it("should have button", () => {
    console.log(wrapper);
    expect(wrapper.find("button")).toHaveLength(1);
  });

  afterEach(() => {
    cleanup();
  });
});
