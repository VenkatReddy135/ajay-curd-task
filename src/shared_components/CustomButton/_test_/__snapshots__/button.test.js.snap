// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Testing renders component 1`] = `
<button
  className=""
  onClick={[Function]}
>
  
</button>
`;
